Realm/RLMThreadSafeReference.h

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+@class RLMRealm;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An instance which is bound to a thread-specific `RLMRealm` instance, and so cannot be passed
+ between threads without being explicitly exported and imported.
+
+ Instances conforming to this protocol can be converted to a thread-safe reference for transport
+ between threads by passing to the `+[RLMThreadSafeReference referenceWithThreadConfined:]`
+ constructor.
+
+ Note that only types defined by Realm can meaningfully conform to this protocol, and defining new
+ classes which attempt to conform to it will not make them work with `RLMThreadSafeReference`.
+ */
+@protocol RLMThreadConfined <NSObject>
+// Conformance to the `RLMThreadConfined_Private` protocol will be enforced at runtime.
+
+/// The Realm which manages the object, or `nil` if the object is unmanaged.
```

Comment:
Maybe "Unmanaged objects are not confined to a thread and cannot be passed to methods expecting a `RLMThreadConfined` object."